space-framework/docs/space-framework.md

```diff
@@ -0,0 +1,92 @@
+# Space Framework
+
+## Terminology 
+**Physical Space**, **pSpace** An entity that behaves like a Kubernetes kube-apiserver (including the persistent storage behind it) and the subset of controllers in the kube-controller-manager that are concerned with API machinery generalities (without the management of containerized workloads). A kcp logical cluster is an example. A regular Kubernetes cluster is another example.
+
+**Space Object**, **Space** - A kind of Kubernetes API object that _represents_ a pSpace.  Its "kind" is `Space`, we sometimes include the "object" part when writing specifically about the API objects to clearly distinguish them from what they represent.
+
+**Space Provider** - A Thing that manages the life cycle of multiple pSpaces. This includes at minimal create/delete/List operations on pSpaces. KIND is an example for Space Provider (that manages KIND clusters), KCP is another example - KCP manages KCP's workspaces. 
+
+## High level architecture  & Main components
+![SF overall view](SF-All.drawio.svg "SF overall view")
+
+The Space Framework (SF) is a generic management framework for space providers and pSpaces. The framework defines an abstraction layer for space providers and pSpace management that allows clients (both CLI/script based and client-go based clients) to use spaces while maintaining the clients decoupled from the specific pSpace and space provider that is being used. 
+
+The space framework uses two CRDs:
+- **SpaceProviderDesc:** This CRD represents a space provider it holds the information needed in order to interact with a specific Space  provider. 
+- **Space:** This CRD represents a space as defined above. It includes reference to secrets that allow clients to connect to the pSpace that it represents and also several attributes used by the SF that will be described later on. 
+
+Spaces can be divided into two categories:
+1. Provider based spaces: The Space Manager (SM) uses a space provider to get information (or perform operations) on the spaces. We can think of such pSpaces as "belong" to the space provider. There are two types of spaces in this category which control the way these spaces are added to the space framework: 
```

Comment:
s/belong/belonging/

Reply:
Why was this marked as resolved?

Reply:
Because it was fixed. Sometimes git still shows old code in the browser. not sure why.

Reply:
last week commit 19ec68eff83b5d829c7a0d2158f3efeb43a98970

Reply:
thanks